src/Doctrine/Annotation/DocLexer.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Doctrine\Annotation;
+
+use PhpCsFixer\Preg;
+
+/**
+ * @internal
+ */
+final class DocLexer
+{
+    public const T_NONE = 1;
+    public const T_INTEGER = 2;
+    public const T_STRING = 3;
+    public const T_FLOAT = 4;
+
+    // All tokens that are also identifiers should be >= 100
+    public const T_IDENTIFIER = 100;
+    public const T_AT = 101;
+    public const T_CLOSE_CURLY_BRACES = 102;
+    public const T_CLOSE_PARENTHESIS = 103;
+    public const T_COMMA = 104;
+    public const T_EQUALS = 105;
+    public const T_FALSE = 106;
+    public const T_NAMESPACE_SEPARATOR = 107;
+    public const T_OPEN_CURLY_BRACES = 108;
+    public const T_OPEN_PARENTHESIS = 109;
+    public const T_TRUE = 110;
+    public const T_NULL = 111;
+    public const T_COLON = 112;
+    public const T_MINUS = 113;
+
+    /** @var array<string, int> */
+    private array $noCase = [
+        '@' => self::T_AT,
+        ',' => self::T_COMMA,
+        '(' => self::T_OPEN_PARENTHESIS,
+        ')' => self::T_CLOSE_PARENTHESIS,
+        '{' => self::T_OPEN_CURLY_BRACES,
+        '}' => self::T_CLOSE_CURLY_BRACES,
+        '=' => self::T_EQUALS,
+        ':' => self::T_COLON,
+        '-' => self::T_MINUS,
+        '\\' => self::T_NAMESPACE_SEPARATOR,
+    ];
+
+    /** @var list<Token> */
+    private array $tokens = [];
+
+    private int $position = 0;
+
+    private int $peek = 0;
+
+    private ?string $regex;
+
+    public function setInput(string $input): void
+    {
+        $this->tokens = [];
+        $this->reset();
+        $this->scan($input);
+    }
+
+    public function reset(): void
+    {
+        $this->peek = 0;
+        $this->position = 0;
+    }
+
+    public function peek(): ?Token
+    {
+        if (isset($this->tokens[$this->position + $this->peek])) {
+            return $this->tokens[$this->position + $this->peek++];
+        }
+
+        return null;
+    }
+
+    /**
+     * @return list<string>
+     */
+    private function getCatchablePatterns(): array
+    {
+        return [
+            '[a-z_\\\][a-z0-9_\:\\\]*[a-z_][a-z0-9_]*',
+            '(?:[+-]?[0-9]+(?:[\.][0-9]+)*)(?:[eE][+-]?[0-9]+)?',
+            '"(?:""|[^"])*+"',
+        ];
+    }
+
+    /**
+     * @return list<string>
+     */
+    private function getNonCatchablePatterns(): array
+    {
+        return ['\s+', '\*+', '(.)'];
+    }
+
+    private function getType(string &$value): int
```

Comment:
```suggestion
    /**
     * @return self::T_*
     */
    private function getType(string &$value): int
```

Reply:
Done, I haven't committed this because the change required an update in 2 places: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7263/commits/3d9de22ebacaafa68dc7eb355dba4ea9a6dc26df